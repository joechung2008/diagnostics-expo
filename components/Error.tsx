import { useTheme } from '@/components/ThemeContext';
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

interface ErrorProps {
  message: string;
}

const Error = ({ message }: ErrorProps) => {
  const { colors } = useTheme();

  const styles = StyleSheet.create({
    container: {
      alignItems: 'center',
      flex: 1,
      justifyContent: 'center',
    },
    errorText: {
      color: colors.error,
      fontSize: 16,
      textAlign: 'center',
    },
  });

  return (
    <View style={styles.container}>
      <Text style={styles.errorText}>{message}</Text>
    </View>
  );
};

export default Error;
