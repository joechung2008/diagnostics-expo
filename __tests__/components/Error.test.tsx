import Error from '@/components/Error';
import { render } from '@testing-library/react-native';
import React from 'react';

jest.mock('@/components/ThemeContext', () => ({
  useTheme: () => ({
    colors: {
      background: '#ffffff',
      surface: '#f5f5f5',
      text: '#000000',
      error: '#ff0000',
    },
  }),
}));

describe('Error component', () => {
  it('renders the error message and matches snapshot', () => {
    const { toJSON } = render(<Error message="Something went wrong" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
