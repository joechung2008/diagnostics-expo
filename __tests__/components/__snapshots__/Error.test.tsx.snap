// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component renders the error message and matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "#ff0000",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    Something went wrong
  </Text>
</View>
`;
